### PR TITLE
[14.0][IMP] t3995 cetmix_tower_server: references in tags

### PR DESCRIPTION
**Pull request by dessanhemrayev**
This commit adds references to tags for easier export/import and action referencing.

Changes
- Inherit from cx.tower.reference.mixin
- Remove field
- Add test
- Add reference fields to tag views, including search view
- Update documentation to reflect new reference fields in tags

Task: 3995

### CODE REVIEW

File: cetmix_tower_server/tests/test_reference_mixin.py [hunk 1]
```diff
@@ -92,3 +92,16 @@ def test_search_by_reference(self):
         # Search using malformed (case sensitive)
         search_result = self.ServerTemplate.get_by_reference("not_much_template")
         self.assertEqual(len(search_result), 0, "Result should be empty")
+
+    def test_search_tag_by_reference(self):
```
Comment (ivs-cetmix) [inline cetmix_tower_server/tests/test_reference_mixin.py]: This function is inherited from mixin and is already tested there. We are not modifying it, so no need to add tests for it too.

**Review by ivs-cetmix — CHANGES_REQUESTED** [review]
Code is good, just no need for tests here)

File: cetmix_tower_server/tests/test_reference_mixin.py (hunk 1 above)
Comment (dessanhemrayev) [reply cetmix_tower_server/tests/test_reference_mixin.py]: Thanks, I'll remove it

**Comment by CetmixGitDrone** [comment]
Congratulations, PR rebased to [14.0-dev](https://github.com/cetmix/cetmix-tower/commits/14.0-dev).

**Pull request closed without merging**